app/directives/proposals/proposals.directive.js

```diff
@@ -137,104 +124,62 @@ angular.module('dashboard')
                 }
             }
 
-            function postProposal(proposal) {
-                $log.debug("dbProposals: postProposal: " + JSON.stringify(proposal));
-                if (proposal instanceof Object) {
-                    var postResult = null;
-
-                    delete proposal.isNew;
-                    delete proposal.isEditing;
-
-                    AhjoProposalsSrv.post(proposal).$promise.then(function(response) {
-                        $log.debug("dbProposals: post then: " + JSON.stringify(response));
-                        postResult = (response instanceof Object) ? response.Data : null;
-                    }, function(error) {
-                        $log.error("dbProposals: post error: " + JSON.stringify(error));
-                    }, function(notify) {
-                        $log.debug("dbProposals: post notify: " + JSON.stringify(notify));
-                    }).finally(function() {
-                        $log.debug("dbProposals: post finally: ");
-                        if (postResult instanceof Object) {
-                            getProposals($scope.guid);
-                        }
-                    });
-                }
-                else {
-                    $log.error('dbProposals: postProposal parameter invalid');
-                }
-            }
-
-            function addProposal(item) {
-                if (angular.isObject(item)) {
-                    if ((self.proposals instanceof Array) === false) {
+            function addProposal(type) {
+                $log.debug("dbProposals: addProposal: " + type);
+                if (angular.isObject(type)) {
+                    if (!angular.isArray(self.proposals)) {
                         self.proposals = [];
                     }
-                    if (item.isModified) {
-                        self.proposals.splice(0, 0, item);
-                    }
-                    else {
-                        self.proposals.splice(0, 0, createDraft(item.value));
-                    }
+                    var draft = createDraft(type.value);
+                    self.proposals.splice(0, 0, draft);
                 }
                 else {
                     $log.error('dbProposals: addProposal parameter invalid');
                 }
                 countProposals();
             }
 
-            function removeProposal(proposal) {
-                $log.debug("dbProposals: removeProposal: " + JSON.stringify(proposal));
-                for (var index = self.proposals.length + CONST.NOTFOUND; angular.isArray(self.proposals) && index > CONST.NOTFOUND; index--) {
+            function removeProposal(guid) {
+                var search = angular.isArray(self.proposals);
+                for (var index = self.proposals.length + CONST.NOTFOUND; search && index > CONST.NOTFOUND; index--) {
```

Comment:
Looks fine, but FYI there's `$rootScope.objWithVal` utility function which does a similar find loop and returns the first matching object.